build/cluster.js

```diff
@@ -76,9 +103,38 @@ gulp.task('spawn-cluster', function(doneFn) {
 });
 
 /**
+ * Checks periodically if heapster is up and running.
+ */
+gulp.task('wait-for-heapster', function(doneFn) {
+  let counter = 0;
+  if (!isHeapsterRunningSetIntervalHandler) {
+    isHeapsterRunningSetIntervalHandler = setInterval(isRunning, 1000);
+  }
+
+  function isRunning() {
+    if (counter % 10 === 0) {
+      gulpUtil.log(
+          gulpUtil.colors.magenta(
+              `Waiting for a Heapster on ${conf.backend.heapsterServerHost}...`));
+    }
+    counter += 1;
+
+    // constantly query the heapster until it is properly running
+    heapsterHealthCheck(function(result) {
+      if (result.length > 0) {
+        gulpUtil.log(gulpUtil.colors.magenta('Heapster is up and running.'));
+        clearTimeout(isHeapsterRunningSetIntervalHandler);
+        isHeapsterRunningSetIntervalHandler = null;
+        doneFn();
+      }
+    });
+  }
+});
+
+/**
  * Checks periodically if cluster is up and running.
  */
-gulp.task('wait-for-cluster', function(doneFn) {
+gulp.task('wait-for-cluster', ['wait-for-heapster'], function(doneFn) {
```

Comment:
Should wait-for-cluster only start after wait-for-heapster is done? I dont think so. The tasks look like independent ones and can be separated. 

If you need one task, then you can create a task wait-for-FOO which depends on these two.